src/nvim/globals.h

```diff
@@ -434,67 +434,74 @@ EXTERN int did_check_timestamps INIT(= FALSE);      /* did check timestamps
-  , HLF_CONCEAL     /* Concealed text */
-  , HLF_SC          /* Sign column */
-  , HLF_SPB         /* SpellBad */
-  , HLF_SPC         /* SpellCap */
-  , HLF_SPR         /* SpellRare */
-  , HLF_SPL         /* SpellLocal */
-  , HLF_PNI         /* popup menu normal item */
-  , HLF_PSI         /* popup menu selected item */
-  , HLF_PSB         /* popup menu scrollbar */
-  , HLF_PST         /* popup menu scrollbar thumb */
-  , HLF_TP          /* tabpage line */
-  , HLF_TPS         /* tabpage line selected */
-  , HLF_TPF         /* tabpage line filler */
-  , HLF_CUC         /* 'cursurcolumn' */
-  , HLF_CUL         /* 'cursurline' */
-  , HLF_MC          /* 'colorcolumn' */
-  , HLF_COUNT       /* MUST be the last one */
+  HLF_8 = 0         // Meta & special keys listed with ":map", text that is
+                    // displayed different from what it is
+  , HLF_EOB         // after the last line in the buffer
+  , HLF_TERM        // terminal cursor focused
+  , HLF_TERMNC      // terminal cursor unfocused
+  , HLF_AT          // @ characters at end of screen, characters that
+                    // don't really exist in the text
+  , HLF_D           // directories in CTRL-D listing
+  , HLF_E           // error messages
+  , HLF_I           // incremental search
+  , HLF_L           // last search string
+  , HLF_M           // "--More--" message
+  , HLF_CM          // Mode (e.g., "-- INSERT --")
+  , HLF_N           // line number for ":number" and ":#" commands
+  , HLF_CLN         // current line number
+  , HLF_R           // return to continue message and yes/no questions
+  , HLF_S           // status lines
+  , HLF_SNC         // status lines of not-current windows
+  , HLF_C           // column to separate vertically split windows
+  , HLF_T           // Titles for output from ":set all", ":autocmd" etc.
+  , HLF_V           // Visual mode
+  , HLF_VNC         // Visual mode, autoselecting and not clipboard owner
+  , HLF_W           // warning messages
+  , HLF_WM          // Wildmenu highlight
+  , HLF_FL          // Folded line
+  , HLF_FC          // Fold column
+  , HLF_ADD         // Added diff line
+  , HLF_CHD         // Changed diff line
+  , HLF_DED         // Deleted diff line
+  , HLF_TXD         // Text Changed in diff line
+  , HLF_CONCEAL     // Concealed text
+  , HLF_SC          // Sign column
+  , HLF_SPB         // SpellBad
+  , HLF_SPC         // SpellCap
+  , HLF_SPR         // SpellRare
+  , HLF_SPL         // SpellLocal
+  , HLF_PNI         // popup menu normal item
+  , HLF_PSI         // popup menu selected item
+  , HLF_PSB         // popup menu scrollbar
+  , HLF_PST         // popup menu scrollbar thumb
+  , HLF_TP          // tabpage line
+  , HLF_TPS         // tabpage line selected
+  , HLF_TPF         // tabpage line filler
+  , HLF_CUC         // 'cursurcolumn'
+  , HLF_CUL         // 'cursurline'
+  , HLF_MC          // 'colorcolumn'
+  , HLF_LS          // LiveSub
```

Comment:
I meant hard coded as `syn_check_group((char_u *)"LiveSub", 7)` instead of using `hl_attr(HLF_LS)` (which can't be done without changing bufhl). If we don't use `HLF_LS` there is no "need" for loops to run through it.